ethcore/sync/src/chain/supplier.rs

```diff
@@ -232,6 +219,24 @@ impl SyncSupplier {
 		Ok(Some((BlockHeadersPacket.id(), rlp)))
 	}
 
+	/// Respond to GetPooledTransactions request
+	fn return_pooled_transactions(io: &dyn SyncIo, r: &Rlp, peer_id: PeerId) -> RlpResponseResult {
+		const LIMIT: usize = 256;
+
+		let transactions = r.iter().take(LIMIT).filter_map(|v| {
+			v.as_val::<H256>().ok().and_then(|hash| io.chain().pooled_transaction(hash))
```

Comment:
As mentioned earlier I think the method should support requesting a batch of transactions from the pool instead of going one-by-one.